blockprod/src/lib.rs

```diff
@@ -89,3 +91,70 @@ pub async fn make_blockproduction(
 
     Ok(Box::new(result))
 }
+
+#[allow(clippy::unused_async, dead_code)]
+async fn setup_blockprod_test() -> (Manager, Arc<ChainConfig>, ChainstateHandle, MempoolHandle) {
```

Comment:
If this is only used for testing, can we guard it by `#[cfg(test)]` and move the in-memory storage dependency to `dev-dependencies`?

Reply:
Yep, I've managed to find a way to move it all into `#[cfg(test)]`. Just tidying up my PR now